### PR TITLE
Week 1 assessment thomas

### PR DESCRIPTION
**Pull request by oddtomas**

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -14,14 +14,38 @@ var temp1 = 35
...
+    } else if (num === 212) {
+        console.log(`${num} is at boiling point`);
+    } else if (num >= 213) {
+        console.log(`${num} is above boiling point`);
+    } else {
+        return "Please enter a number"; //edges in case a number is not entered (for practice)
+    }
+}
+//test
+console.log(boilingTemperature(temp3));
```
Comment (WalkerTexas-coder) [inline code-challenges.js:32]: Good work here!

File: code-challenges.js [hunk 2]
```diff
@@ -14,14 +14,38 @@ var temp1 = 35
...
 // --------------------2) Create a function that takes in two arrays of numbers and returns a single array with all the values sorted least to greatest.
 // Use the test variables provided below. Expected output: [-9, -7, 0, 3, 7, 8, 9, 13, 22, 36]
 
 var myNumbers1 = [3, 7, 0, 36, -9]
 var myNumbers2 = [8, -7, 22, 9, 13]
+var myNumbers3 = []
+//make a function that takes two arrays of numbers
+const arraySorter = () => {
+    myNumbers3 = myNumbers1.concat(myNumbers2).sort((a, b) => a - b) //the sort function by default organizes elements alphabetically so it has to be given more specific instructions to sort numerically. The added instructions tell sort to sort the numbers ascending from least to greatest, to make it descend we could change "a-b" to "b-a". I got the solution from the syllabus but could use help understanding what exactly the a, b represent in this context I understand how it functions but not necessarily what each part represents fully.
```
Comment (WalkerTexas-coder) [inline code-challenges.js:41]: no worries, a and b are placeholders for the two values at different indexs that sort is looking at. Sort is a relly sophisticated algorithm that compares values and moves them around the array. It requires a fair amount of processing power with large arrays.

File: code-challenges.js [hunk 3]
```diff
@@ -30,7 +54,16 @@ var myNumbers2 = [8, -7, 22, 9, 13]
...
+//make a function that takes a string
+const stringReverser = (word) => {
+    var splitString = word.split("") //assigns a new variable the word argument split into an array
+    var reverseString = splitString.reverse() //assigns a new variable with the letters reversed in the array
+    var joinString = reverseString.join("") //joins the reversed array of letters back into a string and removes the quotes
 
+    return joinString //returns the reversed word
+}
+//return the string with characters reversed
+console.log(stringReverser(myString2));
```
Comment (WalkerTexas-coder) [inline code-challenges.js:66]: This is great work here. I like this style of coding because it helps me clearly see what happneing at all steps. This style of code is extrmely easy to refactor down to one line if you feel compelled.

File: code-challenges.js [hunk 4]
```diff
@@ -52,3 +98,17 @@ var numberExampleRefactor2 = "hello"
...
+const subtracterRefracted = (num1, num2) => {
+    //subtract the smaller num from the larger num
+    var result1 = num1 - num2
+    var result2 = num2 - num1
+    if (num1 > num2) { // if the first number is larger it return the results that follows the format of larger - smaller
+        return result1
+    } else if (num2 > num1) {              //else if the second number is larger so it returns num2(larger)-num1(smaller)
+        return result2
+    } else if (typeof num1 !== "number" || typeof num2 !== "number") { //edge case tests for use inputs that are not the typeof "number" and outputs informing the user it's not a valid number
+        console.log("Your input is not a number");
```
Comment (WalkerTexas-coder) [inline code-challenges.js:111]: You can return this last line.

File: coding-concepts.js [hunk 5]
```diff
@@ -8,44 +8,43 @@
...
 // --------------------4) What will this log?
 
 var weekendDays = ["saturday", "sunday"]
-// console.log(weekendDays.toUpperCase())
+//console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will log the array with the strings in capitol letters, because the .toUpperCase() method makes all the letters in a string capitol.
+// b) Verify and explain: I was wrong, the log actually produced an error stating that "weekendDays.toUpperCase is not a function". This is apparently because arrays do not have a toUpperCase method and there is also to toUpperCase function?
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:41]: Correct this string method does not work on arrays. We could use it in the case of itteration though!

File: coding-concepts.js [hunk 6]
```diff
@@ -8,44 +8,43 @@
...
 
 // --------------------5) What will this log?
 
 var dataTypes = ["number", "string", "Boolean"]
-// console.log(typeof dataTypes.length)
+console.log(typeof dataTypes.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This will log the type of data type that the result of dataTypes.length is, which will be a number because .length will evaluate how much data is inside the dataTypes array which will be 3.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:49]: Good attention to detail!

File: interview-questions.md [hunk 7]
```diff
@@ -7,60 +7,58 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: A function is a re-usable block of code, you would use it when you want your code to perform an action like return something or do a math problem etc. Functions are what give websites, apps, etc functionality. Functions make it so that these interfaces are interactable and can do things for the user.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:10]: Yes!

File: interview-questions.md [hunk 8]
```diff
@@ -7,60 +7,58 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 2. In regards to functions, what is the difference between an argument and a parameter?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer: The difference between an parameter and an argument is that the parameter is the data that is expected to be passed to the function and the argument is the actual data that is passed by the user or whoever when the function is invoked/called. When a function is called inside the () braces is where the argument is inputted to be passed to the parameter that's in the layout of the function and from there the function will take that information into consideration when the code is executed.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:18]: Great job showcasing the distictions!

File: interview-questions.md [hunk 9]
```diff
@@ -7,60 +7,58 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: Indexes are sequential numbers that are sort of like addresses for the elements in an array or characters in a string. To access the index we use square brackets with arrays or inside the parenthesis of a built in method for strings. Indexes can be used to change the value at that location by essentially creating a variable with the index position. The length of an array or a string is always the last index position plus one. The content of each element in an array is the actual value, so if an array began with the string "orange" the index would be 0 and the value with be "orange".
 
 
 
 4. What is a built-in method?
 
-  Your answer:
+  Your answer: A built in method is essentially a function that's already built into javascript for things like strings that perform actions that you don't have to define yourself with code. .toUpperCase and .slice are examples of built in methods; they perform actions like upper casing a string or in the case of slice it returns a subset of string with a beginning and ending index. Neither of the methods listed require the user to define the function beforehand you just call it.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:32]: Love the examples!

**Review by WalkerTexas-coder — APPROVED** [review]
Thomas, excellent work! You provided really clean readable code, and you showed that you understood the topics we went over. Thank you for the pseudo code you did put in. I would like to see more of your thought process next week! This was excellent. Your PR is approved!
